airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionUpdaterWorkflow.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling;
+
+import io.airbyte.workers.temporal.scheduling.state.WorkflowState;
+import io.temporal.workflow.QueryMethod;
+import io.temporal.workflow.SignalMethod;
+import io.temporal.workflow.WorkflowInterface;
+import io.temporal.workflow.WorkflowMethod;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@WorkflowInterface
+public interface ConnectionUpdaterWorkflow {
+
+  /**
+   * Workflow method to launch a {@link ConnectionUpdaterWorkflow}. It will launch a workflow
+   * responsible for scheduling the syncs. This workflow will run and then continue as new. So it will
+   * be up until deleted.
+   */
+  @WorkflowMethod
+  void run(ConnectionUpdaterInput connectionUpdaterInput);
+
+  /**
+   * Send a signal that will bypath the waiting time and run a sync. Nothing will happen if a sync is
+   * already running.
+   */
+  @SignalMethod
+  void submitManualSync();
+
+  /**
+   * Cancel all the current executions of a sync and mark the set the status of the job as canceled.
+   * Nothing will happen if a sync is not running.
+   */
+  @SignalMethod
+  void cancelJob();
+
+  /**
+   * Cancel a running workflow and then delete the connection and finally make the workflow to stop
+   * instead of continuing as new
+   */
+  @SignalMethod
+  void deleteConnection();
+
+  /**
+   * Signal that the connection config has been updated. If nothing was currently running, it will
+   * continue the workflow as new, which will reload the config. Nothing will happend if a sync is
+   * running
```

Comment:
> Nothing will happend if a sync is running

Does this mean that if `connectionUpdated()` is called while a sync is running, then even future syncs will not load the new config? Or will any future syncs load the new config before running?

Reply:
It will load the next config. But it won't cancel the sync running (Which can happens on some other operations).